sdk/communication/azure-communication-phonenumbers/README.md

```diff
@@ -103,36 +103,41 @@ The Phone Number Client supports a variety of long running operations that allow
 ### Search for Available Phone Numbers
 Search for available phone numbers by providing the area code, assignment type, phone number capabilities, phone number type, and quantity. The result of the search can then be used to purchase the numbers. Note that for the toll-free phone number type, providing the area code is optional.
 
-<!-- embedme ./src/samples/java/com/azure/communication/phonenumbers/ReadmeSamples.java#L98-L109-->
+<!-- embedme ./src/samples/java/com/azure/communication/phonenumbers/ReadmeSamples.java#L100-L116-->
 ```java
 PhoneNumberCapabilities capabilities = new PhoneNumberCapabilities()
     .setCalling(PhoneNumberCapabilityType.INBOUND)
     .setSms(PhoneNumberCapabilityType.INBOUND_OUTBOUND);
 PhoneNumberSearchOptions searchOptions = new PhoneNumberSearchOptions().setAreaCode("800").setQuantity(1);
 
-PhoneNumberSearchResult searchResult = phoneNumberClient
-    .beginSearchAvailablePhoneNumbers("US", PhoneNumberType.TOLL_FREE, PhoneNumberAssignmentType.APPLICATION, capabilities, searchOptions, Context.NONE)
-    .getFinalResult();
-
-System.out.println("Searched phone numbers: " + searchResult.getPhoneNumbers());
-System.out.println("Search expires by: " + searchResult.getSearchExpiresBy());
-System.out.println("Phone number costs:" + searchResult.getCost().getAmount());
+SyncPoller<PhoneNumberOperation, PhoneNumberSearchResult> poller = phoneNumberClient
+    .beginSearchAvailablePhoneNumbers("US", PhoneNumberType.TOLL_FREE, PhoneNumberAssignmentType.APPLICATION, capabilities, searchOptions, Context.NONE);
+PollResponse<PhoneNumberOperation> response = poller.waitForCompletion();
+String searchId = "";
+
+if (LongRunningOperationStatus.SUCCESSFULLY_COMPLETED == response.getStatus()) {
+    PhoneNumberSearchResult searchResult = poller.getFinalResult();
+    searchId = searchResult.getSearchId();
+    System.out.println("Searched phone numbers: " + searchResult.getPhoneNumbers());
+    System.out.println("Search expires by: " + searchResult.getSearchExpiresBy());
+    System.out.println("Phone number costs:" + searchResult.getCost().getAmount());
+}
 ```
 
 ### Purchase Phone Numbers
 The result of searching for phone numbers is a `PhoneNumberSearchResult`. This can be used to get the numbers' details and purchase numbers by passing in the `searchId` to the purchase number API.
 
-<!-- embedme ./src/samples/java/com/azure/communication/phonenumbers/ReadmeSamples.java#L111-L113 -->
+<!-- embedme ./src/samples/java/com/azure/communication/phonenumbers/ReadmeSamples.java#L118-L120 -->
 ```java
 PollResponse<PhoneNumberOperation> purchaseResponse =
-    phoneNumberClient.beginPurchasePhoneNumbers(searchResult.getSearchId(), Context.NONE).waitForCompletion();
+    phoneNumberClient.beginPurchasePhoneNumbers(searchId, Context.NONE).waitForCompletion();
```

Comment:
Why we need to change this? This seem unrelated to this PR.

Reply:
This code comes from the samples, which is updated